plugins/affinity-group-processors/explicit-dedication/src/main/java/org/apache/cloudstack/affinity/ExplicitDedicationProcessor.java

```diff
@@ -305,6 +305,9 @@ private ExcludeList updateAvoidList(List<DedicatedResourceVO> dedicatedResources
                 for (HostPodVO pod : podList) {
                     DedicatedResourceVO dPod = _dedicatedDao.findByPodId(pod.getId());
                     if (dPod != null && !dedicatedResources.contains(dPod)) {
+                        if (s_logger.isDebugEnabled()) {
+                            s_logger.debug(String.format("Avoiding POD %s [%s] because it is dedicated.", pod.getName(), pod.getUuid()));
```

Comment:
seems the cond is the check if the pod is not in dedicated resources.

```suggestion
                            s_logger.debug(String.format("Avoiding POD %s [%s] because it is not dedicated.", pod.getName(), pod.getUuid()));
```